libpod/networking_linux.go

```diff
@@ -172,9 +169,19 @@ func (r *Runtime) setupRootlessNetNS(ctr *Container) (err error) {
 	havePortMapping := len(ctr.Config().PortMappings) > 0
 	apiSocket := filepath.Join(r.ociRuntime.tmpDir, fmt.Sprintf("%s.net", ctr.config.ID))
 
+	for i, j := range ctr.config.Spec.Linux.Namespaces {
+		if j.Type == "network" {
+			ctr.config.Spec.Linux.Namespaces[i].Path = ctr.state.NetNS.Path()
+		}
+	}
+
+	if err := ctr.saveSpec(ctr.config.Spec); err != nil {
+		logrus.Errorf("Failure saving spec to disk: %+v", err)
+	}
+
 	cmdArgs := []string{}
 	if havePortMapping {
-		cmdArgs = append(cmdArgs, "--api-socket", apiSocket, fmt.Sprintf("%d", ctr.state.PID))
```

Comment:
@giuseppe Do you know the reasoning for appending the container PID with api-socket? I couldn't figure out the reasoning behind it, but some of the port mapping tests are failing and I want to make sure its not because of removing it.

Reply:
@giuseppe ^^